injection/sharedmain/main.go

```diff
@@ -187,36 +200,38 @@ func MainWithConfig(ctx context.Context, component string, cfg *rest.Config, cto
 		<-ctx.Done()
 	}
 
-	// Set up leader election config
-	leaderElectionConfig, err := GetLeaderElectionConfig(ctx)
-	if err != nil {
-		logger.Fatalw("Error loading leader election configuration", zap.Error(err))
-	}
-	leConfig := leaderElectionConfig.GetComponentConfig(component)
-
-	if !leConfig.LeaderElect {
+	if *leaderElectionEnabled {
```

Comment:
what is going on with the `*`? is that required?

Reply:
Yes. Because `flag.Bool(...)` returns a pointer to bool. So we dereference it.